lib/ao_host.js

```diff
@@ -577,18 +572,15 @@ class AOHost extends AsyncEventEmitter {
    * to required channels, and triggers the life.start event.
    *
    * @param {object} instance - algo order instance that has started
-   * @private
    */
-  async onAOStart (instance = {}) {
-    const { gid } = instance.state
+  async _startAlgo (instance = {}) {
+    const { gid, name, label, args } = instance.state
 
     await withAOUpdate(this, gid, (instance = {}) => {
       const { state = {} } = instance
+      state.active = true
 
-      return {
-        ...state,
-        active: true
-      }
+      return false
```

Comment:
`withAOUpdate` is useless since the callback always returns false.
Can be replaced with `instance.state.active = true`?

Reply:
good point, fixed!